src/gt4py/next/iterator/embedded.py

```diff
@@ -470,7 +470,9 @@ def promote_scalars(val: CompositeOfScalarOrField):
     elif isinstance(val, LocatedField):
         return val
     val_type = infer_dtype_like_type(val)
-    if np.issubdtype(val_type, np.number):
+    if np.issubdtype(val_type, np.number) or (
+        isinstance(val_type, type) and issubclass(val_type, bool)
+    ):
```

Comment:
Following https://numpy.org/doc/stable/reference/arrays.scalars.html, I would rewrite the condition in this way:
```suggestion
    if np.issubdtype(val_type, np.number) or (
        np.issubdtype(val_type, np.bool_)
    ):
```
Actually, it might be interesting to factor this out as its own function, and use it everywhere in the embedded implementation. I have the feeling that right now there are several places using slightly wrong variations of this same check.